src/GovUk.Education.ExploreEducationStatistics.Data.Services/ObservationPredicateBuilder.cs

```diff
@@ -233,6 +251,16 @@ private static bool ObservationalUnitExists(LocationQuery query)
                 observation => query.RscRegion.Contains(observation.Location.RscRegion_Code));
         }
 
+        private static Expression<Func<Observation, bool>> SchoolPredicate(LocationQuery query)
+        {
+            var schoolOldCodes = query.School.Where(s => s.Length == 3).ToList();
```

Comment:
Are you sure the laestab is a 3 digit code. This looks like it might be based on the format for an old code.